keanu-python/keanu/algorithm/sampling.py

```diff
@@ -142,21 +145,36 @@ def generate_samples(net: BayesNet,
     samples = samples.dropCount(drop).downSampleInterval(down_sample_interval)
     sample_iterator: JavaObject = samples.stream().iterator()
 
-    print(vertices_unwrapped)
+    all_are_scalar = __all_scalar(sample_from)
+
     return _samples_generator(
-        sample_iterator, vertices_unwrapped, live_plot=live_plot, refresh_every=refresh_every, ax=ax)
+        sample_iterator,
+        vertices_unwrapped,
+        live_plot=live_plot,
+        refresh_every=refresh_every,
+        ax=ax,
+        all_scalar=all_are_scalar)
+
+
+def __all_scalar(sample_from):
+    return all((vertex.get_value().shape == () for vertex in sample_from))
 
 
 def _samples_generator(sample_iterator: JavaObject, vertices_unwrapped: JavaList, live_plot: bool, refresh_every: int,
-                       ax: Any) -> sample_generator_types:
+                       ax: Any, all_scalar: bool) -> sample_generator_types:
     traces = []
     x0 = 0
     while (True):
         network_sample = sample_iterator.next()
-        sample = {
-            Vertex._get_python_label(vertex_unwrapped): Tensor._to_ndarray(network_sample.get(vertex_unwrapped))
-            for vertex_unwrapped in vertices_unwrapped
-        }
+
+        if all_scalar:
+            sample: Dict[Union[str, Tuple[str, str]], primitive_types] = {
+                Vertex._get_python_label(vertex_unwrapped): Tensor._to_ndarray(
```

Comment:
We need a new method for this, because it's not returning an `ndarray`

Reply:
`_to_scalar_or_ndarray`